File: testsuite/communication_stdio_stub/stub.java
import java.util.Scanner;

public class Stub {
  public static void main(String []args) {
    Scanner scan = new Scanner(System.in);
    String command = scan.next();
    String data = scan.next();
    Encoder encoder = new Encoder();
    if (command.equals("ENCODE")) {
      System.out.println(encoder.encode(data));
    }
    else {
      System.out.println(encoder.decode(data));
    }
  }
}

File: testsuite/communication_stdio_stub/stub.cpp
#include <iostream>
#include <string>
#include "header.h"

using namespace std;

int main() {
  string command, data;
  cin >> command >> data;
  if (command == "ENCODE") {
    cout << encode(data) << endl;
  } else {
    cout << decode(data) << endl;
  }
  return 0;
}
